File: app/src/main/java/com/example/myapplication/data/api/AuthApiService.java
package com.example.myapplication.data.api;

import com.example.myapplication.data.model.ApiResponse;
import com.example.myapplication.data.model.LoginRequest;
import com.example.myapplication.data.model.LoginResponse;
import com.example.myapplication.data.model.RegisterRequest;
import com.example.myapplication.data.model.User;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

/**
 * 认证API服务接口
 * renZheng_API_fuWu
 */
public interface AuthApiService {

    /**
     * 用户登录
     * yongHu_dengLu
     */
    @POST("auth/login")
    Call<ApiResponse<LoginResponse>> login(@Body LoginRequest loginRequest);

    /**
     * 用户注册
     * yongHu_zhuCe
     */
    @POST("auth/register")
    Call<ApiResponse<User>> register(@Body RegisterRequest registerRequest);

    /**
     * 刷新token
     * shuaXin_token
     */
    @POST("auth/refresh")
    Call<ApiResponse<LoginResponse>> refreshToken(@Body String refreshToken);

    /**
     * 登出
     * dengChu
     */
    @POST("auth/logout")
    Call<ApiResponse<Void>> logout();

    /**
     * 获取当前用户信息
     * huoQu_dangQian_yongHu_xinXi
     */
    @POST("auth/me")
    Call<ApiResponse<User>> getCurrentUser();
}

File: app/src/main/java/com/example/myapplication/api/AuthApiService.kt
package com.example.myapplication.api

import com.example.myapplication.model.ApiResponse
import com.example.myapplication.model.LoginRequest
import com.example.myapplication.model.LoginResponse
import com.example.myapplication.model.RegisterRequest
import com.example.myapplication.model.UserDto
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApiService {

    @POST("api/auth/login")
    fun login(@Body request: LoginRequest): Call<ApiResponse<LoginResponse>>

    @POST("api/auth/register")
    fun register(@Body request: RegisterRequest): Call<ApiResponse<UserDto>>

    @POST("api/auth/me")
    fun getCurrentUser(): Call<ApiResponse<UserDto>>

    @POST("api/auth/logout")
    fun logout(): Call<ApiResponse<Void>>
}
